src/Microsoft.AspNetCore.Routing/RouteBase.cs

```diff
@@ -46,12 +46,19 @@ public abstract class RouteBase : IRouter, INamedRouter
             ConstraintResolver = constraintResolver;
             DataTokens = dataTokens ?? new RouteValueDictionary();
 
-            // Data we parse from the template will be used to fill in the rest of the constraints or
-            // defaults. The parser will throw for invalid routes.
-            ParsedTemplate = TemplateParser.Parse(template);
+            try
+            {
+                // Data we parse from the template will be used to fill in the rest of the constraints or
+                // defaults. The parser will throw for invalid routes.
+                ParsedTemplate = TemplateParser.Parse(template);
 
-            Constraints = GetConstraints(constraintResolver, ParsedTemplate, constraints);
-            Defaults = GetDefaults(ParsedTemplate, defaults);
+                Constraints = GetConstraints(constraintResolver, ParsedTemplate, constraints);
+                Defaults = GetDefaults(ParsedTemplate, defaults);
+            }
+            catch (Exception exception)
```

Comment:
Can we tighten up the scope of this? For instance - this could rewrap a `RouteCreationException` thrown by `GetConstraints` - is that what we want?